src/wallet/rpcwallet.cpp

```diff
@@ -3672,9 +3673,9 @@ class DescribeWalletAddressVisitor : public boost::static_visitor<UniValue>
             // Always report the pubkey at the top level, so that `getnewaddress()['pubkey']` always works.
             if (subobj.exists("pubkey")) obj.pushKV("pubkey", subobj["pubkey"]);
             obj.pushKV("embedded", std::move(subobj));
-        } else if (which_type == TxoutType::MULTISIG) {
-            // Also report some information on multisig scripts (which do not have a corresponding address).
-            // TODO: abstract out the common functionality between this logic and ExtractDestinations.
+        } else if (find(enabled_methods.begin(), enabled_methods.end(), "reqSigs") != enabled_methods.end() &&
```

Comment:
It's pretty ugly to duplicate this logic in many places. I'd suggest just using `IsDeprecatedRPCEnabled`, in the top-level RPC code, and pass down a boolean to wherever it's needed (e.g. in core_write, make `ScriptPubKeyToUniv` take a boolean to indicate whether to not to include `reqSigs` / `addresses`).

Reply:
I was able to do this how you and @MarcoFalke suggested in this commit https://github.com/bitcoin/bitcoin/pull/20286/commits/dba1053a5e0c74e53ded3d42c3826373474ba805. However, I'm a little confused by what's going on in `rpcwallet.cpp`. I'm able to use `IsDeprecatedRPCEnabled` here when I build and test locally (macOS), and it seems some of the CI test runs pass. However, some of the CI runs throw a compile error in this case. Any ideas or suggestions on the approach I should take for `rpcwallet`?